### PR TITLE
codeintel: Move `committed_at` out of the hot path

### LINKED ISSUE #42997: codeintel: Reduce `committed_at` contention
We should remove the `committed_at` column from on lsif_uploads (and possibly lsif_indexes) records into a shared table which maps repository/commit to a date. This would not require us to update the lsif_upload record while it's processing (which seems to conflict with heartbeat operations), and instead we can write (conflict-free) to a secondary table. Queries will need to join to this new table to read the value.

This table should not be large (on the order of #uploads) and does not store large data. Therefore we don't need to prune it (we don't want to cascade deletes from the job records as it's shared between many jobs for the same commit). We should be mindful of this in the future as we're treating it as an append-only table for the moment.

See [this brainstorming doc](https://docs.google.com/document/d/1wBHmnUjWZ7LrPzIrwbcqcJ7bt0rAPapa_7sWEjWWOSU) for additional context.

### PR DESCRIPTION
**Pull request by efritz**
This PR breaks the `committed_at` dates on `lsif_uploads` into a new table. Fixes #42997.

## Test plan

Tested locally, modified unit tests to match new expected behavior.

### CODE REVIEW

File: migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE IF NOT EXISTS codeintel_commit_dates(
+    repository_id integer NOT NULL,
```
Comment (eseliger) [inline migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql:2]: should this be a foreign key?

File: migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    commit_bytea bytea NOT NULL,
+    committed_at timestamp with time zone,
+    PRIMARY KEY(repository_id, commit_bytea)
+);
```
Comment (eseliger) [inline migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql:6]: might file a ticket to drop that old col later

Comment (eseliger) [reply migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql:6]: would probably migrate existing data, even if it's backfilled

**Review by eseliger — APPROVED** [review]

File: migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql (hunk 2 above)
Comment (efritz) [reply migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql:6]: Updated.

File: migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql (hunk 1 above)
Comment (efritz) [reply migrations/frontend/1665770699_add_codeintel_commit_dates/up.sql:2]: We have a lot of codeintel-scoped stuff that references ids without fk-integrity. I think I want to do a full pass over these and see whether adding fks to non-codeintel tables makes sense in general (for now lsif_uploads have the same implicit relationship to repos).

**Pull request merged**